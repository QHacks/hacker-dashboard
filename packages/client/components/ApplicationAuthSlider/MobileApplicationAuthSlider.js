import React, { Component } from "react";
import { redDark, red, steel } from "../../assets/colors";

class Mobile extends Component {
  changeSelected(index) {
    this.props.changeSelected(index);
  }

  render() {
    const items = this.props.items.map((item, i) => (
      <button
        css={`
          display: block;
          text-align: center;
          line-height: 44px;
          text-transform: uppercase;
          cursor: pointer;
          background: none;
          border: none;
          font-size: 18px;
          z-index: 3;
          transition: 0.5s ease;
          border-radius: 44px;
          width: 100%;
          margin: 12px auto;
          max-width: 300px;
          border: 1px solid ${steel};
          ${this.props.selectedItem === i
            ? `color: white; background-color: ${red};`
            : `color: #4d4d4d; &:hover { color: ${redDark}; }
          `}
        `}
        key={item}
        onClick={() => this.changeSelected(i)}
        type="button"
      >
        {item}
      </button>
    ));

    return (
      <div
        css={`
          margin-top: 35px;
        `}
      >
        {items}
        <hr
          css={`
            color: #c5c9d1;
            margin: 52px auto 10px;
          `}
        />
      </div>
    );
  }
}

export default Mobile;
