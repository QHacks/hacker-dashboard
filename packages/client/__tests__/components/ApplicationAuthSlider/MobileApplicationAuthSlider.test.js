import React from "react";
import MobileApplicationAuthSlider from "../../../components/ApplicationAuthSlider/MobileApplicationAuthSlider";
import { shallow } from "enzyme";

describe("ApplicationAuthSlider", () => {
  it("renders correctly", () => {
    expect(
      shallow(<MobileApplicationAuthSlider items={["TEST"]} />)
    ).toMatchSnapshot();
  });
});
