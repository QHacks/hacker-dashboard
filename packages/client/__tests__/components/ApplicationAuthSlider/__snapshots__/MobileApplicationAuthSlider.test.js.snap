// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationAuthSlider renders correctly 1`] = `
<div
  className="css-wmurvm"
>
  <button
    className="css-1j5jfx0"
    key="TEST"
    onClick={[Function]}
    type="button"
  >
    TEST
  </button>
  <hr
    className="css-k0uucn"
  />
</div>
`;
